en/components/for_of.md

```diff
@@ -107,15 +105,83 @@ The directive can be used to virtualize the data in vertical, horizontal or both
     </ng-template>
 </table>
 ```
+### igxFor bound to remote service
 
-**Note**: The `igxForOf` directive can bind to remote service too. Review the following demo for more information.
-#### igxFor bound to remote service Demo
-<div class="sample-container loading" style="height:780px">
-    <iframe src='{environment:demosBaseUrl}/igx-for-sample-2' width="100%" height="100%" seamless frameBorder="0" onload="onSampleIframeContentLoaded(this);"></iframe>
+The `igxForOf` directive can be bound to remote service. You need to use `Observeble` property - `remoteData`(in the following case). Also the `chunkLoading` event should be utilized to trigger the requests to the data.
+
+```html
+<div style='height:500px;'>
+    <ng-template igxFor let-item [igxForOf]="remoteData | async" (onChunkPreload)="chunkLoading($event)"
+    [igxForScrollOrientation]="'vertical'"
+    [igxForContainerSize]='"500px"'
+    [igxForItemSize]='"50px"'
+    [igxForRemote]='true'
+    let-rowIndex="index" #virtDirRemote>
+    <div style='height:50px;'>{{item.ProductID}} : {{item.ProductName}}</div>
+    </ng-template>
 </div>
-<br/>
-<div class="divider--half"></div>
+```
 
+Also there is a requirement to set the `totatItemsCount` property in the instance of `igxForOf`.
+```typescript
+this.virtDirRemote.totalItemCount = data["@odata.count"];
+```
+
+In order access the directive instance from the component it should be marked as `ViewChild`:
+```typescript
+@ViewChild("virtDirRemote", { read: IgxForOfDirective })
+public virtDirRemote: IgxForOfDirective<any>;
+```
+And after the request to load the first chunk, the `totalItemCount` can be set:
+```typescript
+public ngAfterViewInit() {
+    this.remoteService.getData(this.virtDirRemote.state, (data) => {
+        this.virtDirRemote.totalItemCount = data["@odata.count"];
+    });
+}
+```
+When requesting data you can take advantage of `IgxForOfState` interface, which provides the `startIndex` and `chunkSize`. But note, that initialy the chunkSize will be 0, so you have to specify the size of the first loaded chunk(the best value is `igxForContainerSize` divided on `igxForItemSize`).
```

Comment:
initially
"divided on"?  "divided by"?

Reply:
"the best value is `igxForContainerSize` initially divided by `igxForItemSize` by sounds better I think